src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -352,6 +347,21 @@ func (r *StatefulSetResource) Obj() (k8sclient.Object, error) {
 		},
 	}
 
+	if r.pandaCluster.Spec.Configuration.TLS.KafkaAPIEnabled {
```

Comment:
NOT A BLOCKER
It's a matter of preference, but this could introduce bad precedence. Let's assume we will have multiple modification based on Cluster CR. The external connectivity, volumes in TLS or persistent volume claim change based on storage class. 

I think more readable is to have function that returns `VolumeMounts` and other that returns `Volume` and are inlined with the STS definition. First function should be inline [in the volumemount definition](https://github.com/vectorizedio/redpanda/pull/729/files#diff-11cd0daf9c52813e733b2526ffd440b85ff6e9a8f6a11d8454c02cbd129bcb60R288-R297) by appending to the always available mounts.

Example https://github.com/vectorizedio/redpanda/pull/694/files#diff-11cd0daf9c52813e733b2526ffd440b85ff6e9a8f6a11d8454c02cbd129bcb60R307-R316

Reply:
I happen to be working around the StorageClass (in a separate PR) and I'm already creating such a function, so I'll take care of it there.

Reply:
Ok